# Prüfungsleitung Maschnielles Lernen: Gruppe 8

Isolet download: https://datahub.io/machine-learning/isolet

### Aufgabe 1

#### 1) Data Loading and Splitting 

In [1]:
# Import isolet.csv File
import pandas as pd

# import
df_raw_isolet = pd.read_csv("./raw_isolet.csv", delimiter=",")

##### Einfache Datenbetrachtung

In [2]:
print("Einträge in isolet.csv", len(df_raw_isolet))
print("Gruppierte Einträge:", df_raw_isolet.groupby(['class']).size())

Einträge in isolet.csv 7797
Gruppierte Einträge: class
'1'     300
'10'    300
'11'    300
'12'    300
'13'    299
'14'    300
'15'    300
'16'    300
'17'    300
'18'    300
'19'    300
'2'     300
'20'    300
'21'    300
'22'    300
'23'    300
'24'    300
'25'    300
'26'    300
'3'     300
'4'     300
'5'     300
'6'     298
'7'     300
'8'     300
'9'     300
dtype: int64


Bei der Betrachtung der Anzahl ist aufällig, dass in der Regel pro Buchstabe 300 Einträge aufgezeichnet wurden. Da aber len(df) nur 7797 Einträge beinhaltet, gibt es einige Ausnahmen. <br>
Ausnahmen: Buchstabe 13 mit 299 Einträgen und Buchstabe 6 mit 298 Einträgen <br><br>
Insgesamt bietet die sehr ausgewogene Verteilung der jeweiligen Buchstaben eine sehr gute Basis dies in dem Data Splitting zu betrachten. So wird für jeden Buchstaben ein Testsatz von je 75 Einträhge bestimmt. Formel: ⎡Anzahl Einträge pro Buchstabe * 0,25⎤

##### Data Splitting

In [ ]:
df_letters_training = 'todo'
df_letters_test = 'todo'

#### 2) Merkmalsreduktion

#### 3) Implementierung der Klassifikatoren

#### 4) Evaluation